components/centraldashboard/public/components/activity-view.js

```diff
@@ -1,112 +1,109 @@
-                    padding: 0 5px;
-                }
-                .event {
-                    flex-basis: 50%;
-                }
-                .event-icon {
-                    height: 18px;
-                    width: 18px;
-                    padding: 0 10px;
-                }
-                .event-icon.build {
-                    transform: scale(-1, 1);
-                    color: rgb(28, 175, 199);
-                }
-                .event-icon.error {
-                    color: rgb(218, 66, 54);
-                }
                 [hidden] {
                     display: none;
                     opacity: 0;
                     pointer-events: none
                 }
             </style>
-            <iron-ajax auto url="/api/activities" handle-as="json"
-                loading="{{loading}}" on-response="_onResponse">
+            <iron-ajax id="ajax" url="/api/activities/[[namespace]]"
+                handle-as="json" loading="{{loading}}"
+                on-response="_onResponse">
             </iron-ajax>
             <paper-progress indeterminate class="slow"
                 hidden$="[[!loading]]"></paper-progress>
-            <div id="list" role="listbox">
-                <template is="dom-repeat" items="[[activities]]">
-                    <div class="activity-row layout horizontal">
-                        <div class="flex">[[item.formattedTime]]</div>
-                        <div class="flex flex-auto event">
-                            <iron-icon class$="event-icon [[item.icon]]"
-                                icon="[[item.icon]]">
-                            </iron-icon>[[item.event]]
-                        </div>
-                        <div class="flex">[[item.source]]</div>
-                    </div>
-                </template>
-            </div>
+            <p hidden$="[[!message]]">[[message]]</p>
+            <activities-list activities="[[activities]]"></activities-list>
             `;
     }
 
     /**
-      * Object describing property-related metadata used by Polymer features
-      */
+     * Object describing property-related metadata used by Polymer features
+     */
     static get properties() {
         return {
-            loading: Boolean,
-            activities: Array,
+            namespace: {
+                type: String,
+                observer: '_namespaceChanged',
+                value: null,
+            },
+            message: {
+                type: String,
+                value: 'Select a namespace to see recent events',
+            },
+            loading: {
+                type: Boolean,
+                value: false,
+            },
+            activities: {
+                type: Array,
+                value: [],
+            },
         };
     }
 
+    /**
+     * Retrieves Events when namespace is selected.
+     * @param {string} newNamespace
+     */
+    _namespaceChanged(newNamespace) {
+        if (newNamespace) {
```

Comment:
Invert this condition to decrease code depth:
```suggestion
        if (!newNamespace) return;
        this.message = '';
        this.$['ajax'].generateRequest();
```

Reply:
I've generally followed the "positive condition first" as I find it to be more readable, but I'm fine making this change.